### PR TITLE
Feature/user settings

### PR DESCRIPTION
**Pull request by finye**
User settings without interest part

### CODE REVIEW

File: client/src/pages/UserSettings.tsx [hunk 1]
```diff
@@ -47,18 +50,56 @@ const LanguageButtonFin = styled(Button)(() => ({
...
+    }
+  }, [user])
+
+  const fetchedUserData = useGetUserData(userSub)
+
+  useEffect(() => {
+    if(fetchedUserData) {
+      setUserData(fetchedUserData)
+    }
+  }, [fetchedUserData])
```
Comment (finye) [inline client/src/pages/UserSettings.tsx:79]: We need to make sure that we are doing this once or only until we get the user data. Passing fetchedUserData as it is might lead to uncontrolled state management. Can we please wrap this with useCallback or some optimizer like useMemo

File: client/src/pages/UserSettings.tsx [hunk 2]
```diff
@@ -47,18 +50,56 @@ const LanguageButtonFin = styled(Button)(() => ({
...
+    interests:  [] as (string | null)[],
   }
 
   const [formData, setFormData] = useState(initialUserSetting)
 
+  useEffect(() => {
+    const userItem = userData?.userBySub?.items[0]
+    if (userItem) {
+      const { givenName, familyName, email, language, interestTags } = userItem;
+      setFormData({
```
Comment (finye) [inline client/src/pages/UserSettings.tsx:97]: We probably don't need to use setFormData here. For the initial stage of the user settings page, we can add the data we get from `userData` and replace the values in `initialUserSetting`. `setFormData` should only be updated by the input or setting changes done by the user

Comment (finye) [reply client/src/pages/UserSettings.tsx:97]: We can also possibly minimize the amount of useEffects

**Review by finye — COMMENTED** [review]
Note to self and @Zantemann 😄

File: client/src/hooks/appSyncHooks.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react'
+import { API, GraphQLQuery } from '@aws-amplify/api'
+//import { userBySub } from '../graphql/queries'
+import * as queries from '../graphql/generatedQueries'
+import { UserBySubQuery } from '../types/generatedAPI';
+
+export const useGetUserData = (userSub: string) => {
```
Comment (finye) [inline client/src/hooks/appSyncHooks.tsx:7]: We can keep this hook and why not use `useAuthUser()` here, instead of passing it as a parameter?

Comment (Zantemann) [reply client/src/hooks/appSyncHooks.tsx:7]: Yeah, we can use it there

Comment (finye) [reply client/src/hooks/appSyncHooks.tsx]: Great 👍

**Comment by Zantemann** [comment]
Translations missing. Message after successful or failed save to user? Set and update language according to user settings, maybe another issue.

**Comment by finye** [comment]
> Translations missing. Message after successful or failed save to user? Set and update language according to user settings, maybe another issue.

Yes, I will work on that 👍

**Review by Zantemann — APPROVED** [review]
Everything working

**Pull request merged**